bin/improver-weighted-blending

```diff
@@ -46,63 +47,86 @@ def main():
        Then load in the data to blend and calculate default weights
        using the method chosen before carrying out the blending."""
     parser = ArgParser(
-    description='Calculate the default weights to apply in weighted ' +
-                'blending plugins using the ChooseDefaultWeightsLinear or ' +
-                'ChooseDefaultWeightsNonLinear plugins. Then apply these ' +
-                'weights to the cube using the BasicWeightedAverage plugin.' +
-                ' Required for ChooseDefaultWeightsLinear: y0val and ONE ' +
-                'of slope, ynval. Required for ' +
+    description='Calculate the default weights to apply in weighted '
+                'blending plugins using the ChooseDefaultWeightsLinear or '
+                'ChooseDefaultWeightsNonLinear plugins. Then apply these '
+                'weights to the cube using the BasicWeightedAverage plugin.'
+                ' Required for ChooseDefaultWeightsLinear: y0val and ONE '
+                'of slope, ynval. Required for '
                 'ChooseDefaultWeightsNonLinear: cval.')
     group = parser.add_mutually_exclusive_group(required=True)
     group.add_argument('--linear', default=False, action='store_true',
```

Comment:
I've changed this to use one positional argument `wts_calc_method` with 2 choices (`linear` and `nonlinear` instead. I had to tweak most of the tests also.